l1-contracts/src/core/ProofCommitmentEscrow.sol

```diff
@@ -0,0 +1,153 @@
+// SPDX-License-Identifier: Apache-2.0
+// Copyright 2024 Aztec Labs.
+pragma solidity >=0.8.27;
+
+import {SafeERC20} from "@oz/token/ERC20/utils/SafeERC20.sol";
+import {IERC20} from "@oz/token/ERC20/IERC20.sol";
+
+import {IProofCommitmentEscrow} from "@aztec/core/interfaces/IProofCommitmentEscrow.sol";
+import {Errors} from "@aztec/core/libraries/Errors.sol";
+import {Constants} from "@aztec/core/libraries/ConstantsGen.sol";
+
+contract ProofCommitmentEscrow is IProofCommitmentEscrow {
+  using SafeERC20 for IERC20;
+
+  struct Stake {
+    uint256 amount;
+    address prover;
+  }
+
+  struct WithdrawRequest {
+    uint256 amount;
+    uint256 executableAt;
+  }
+
+  address public immutable OWNER;
+  uint256 public constant WITHDRAW_DELAY =
+    Constants.ETHEREUM_SLOT_DURATION * Constants.AZTEC_EPOCH_DURATION * 3;
+  mapping(address => uint256) public deposits;
+  mapping(address => WithdrawRequest) public withdrawRequests;
+  IERC20 public token;
+  Stake public stake;
+
+  modifier onlyOwner() {
+    if (msg.sender != OWNER) {
+      revert Errors.ProofCommitmentEscrow__NotOwner(msg.sender);
+    }
+    _;
+  }
+
+  modifier hasBalance(address _prover, uint256 _amount) {
```

Comment:
Nit: Stylewise, Im generally not a fan of using modifiers to perform anything other than an address check, I find it more clear when this check is explicitly written into a function.